### ISSUE TITLE
Two factor code input should use keypad on mobile devices 

### ISSUE THREAD

**Issue by noiob** [tootsuite/mastodon#5527]
When entering the two-factor code on my phone, it's always a hassle to switch to the number part of the keyboard. Setting the input type / constraints right should allow the browser to show a number pad automatically.

According to [this stackoverflow answer](https://stackoverflow.com/questions/6178556/phone-numeric-keyboard-for-text-input#answer-31619311), setting `<input type="number" pattern="[0-9]*" inputmode="numeric">` should work on iOS and Android, but it's from 2015, so there might be a better way.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by nolanlawson** [comment]
Is it guaranteed that every OTP app uses numbers only? I've tried FreeOTP and Google Authenticator and both use numbers AFAICR.

**Comment by noiob** [comment]
Since they all implement the same standard, yes

**Comment by nightpool** [comment]
@noiob not sure we can implement this—entering recovery codes requires the ability to type a 16-digit hex-encoded string (not just numbers)

**Comment by noiob** [comment]
Good point. Maybe add a button "enter recovery code" that just changes the input type? It's not clear that recovery codes can just be entered into the same field anyways

**Comment by nightpool** [comment]
I'd totally be down for that, although it's more non-react js

**Comment by noiob** [comment]
I'm sure there's a react way to do it, and it should probably be done that way

**Comment by nolanlawson** [comment]
See discussion in #5561; I think it makes the most sense to just stick with plaintext.

@noiob Out of curiosity, does your OTP app not allow you to copy the code to a clipboard? I've only ever pasted in the 2FA code rather than typing it in.

**Comment by noiob** [comment]
@nolanlawson I use [Authenticator](http://apps.getpebble.com/en_US/application/52f1a4c3c4117252f9000bb8?query=authenticator&section=watchapps) on my Pebble 2 smartwatch, so copy-and-paste is not an option. Maybe my workflow isn't the best, but my phone isn't the best at multitasking anyways.
